### ISSUE TITLE
Qtile doesn't start with python-cairocffi-1.6.0 and python-xcffib-1.4.0

### ISSUE THREAD

**Issue by leemon22** [qtile/qtile#4491]
### Issue description

I updated sone packages with `sudo pacman -Syu`. Some of these packages are lib32-glibc, libx11, lib32-libx11, libxpm, npm, python-cairocffi, python-xcffib.
Then I restarted my computer.
I logged in with my greeter, which is lightdm, and the screen goes black and I go back to the greeter.
I checked the problem is no related to only x11 installing other wm like i3, and  it works.

### Version

0.22.1

### Backend

X11 (default)

### Logs

```text
2023-10-04 09:47:51,689 ERROR libqtile manager.py:_finalize_configurables():L301 exception during finalize
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 289, in _finalize_configurables
    widget.finalize()
  File "/usr/lib/python3.11/site-packages/libqtile/widget/systray.py", line 289, in finalize
    self.conn.conn.flush()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 581, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2023-10-04 09:47:51,692 ERROR libqtile core.py:_xpoll():L332 Shutting down due to X connection error XCB_CONN_ERROR (1)
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 294, in _xpoll
    event = self.conn.conn.poll_for_event()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
```

### Required

- [X] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by elParaguayo** [comment]
That's odd. Starts fine for me.

I suspect there's a different issue somewhere. That log doesn't look complete either - the first line is about finalising widgets which would happen once qtile has started shutting down. Can you post (a link to) your full log so we can take a look?

**Comment by leemon22** [comment]
After opening this issue, I have done the following:

1. I tested if other window managers worked. I tried i3 and awesome, no issues.
2. Then I tried the following:
2.1. I deleted old logs.
2.2. I backed up my configurations.
2.3. I deleted my configurations.
2.4. I reinstalled qtile with the goal of starting fresh.
3. I tried to start qtile again, but the same error keeps occurring.
4. I checked the configuration file with `cat ~/.local/share/qtile/qtile.log, but it says the file doesn't exist.

I didn't expect this, for qtile to fail and not leave any logs.
I've attached a video of what's happening.You can see that the program I use for notifications is working. I don't know what to do XD.
https://www.youtube.com/watch?v=W_b7EuMO85g

**Comment by randomcodepanda** [comment]
I also failed to start qtile after today's upgrade, i downgraded python-cairocffi and python-xcffib and then it worked again.
But now i'm intrigued since apparently ElParaguayo was able to upgrade and run it without any problems.

**Comment by elParaguayo** [comment]
Ah! That's what's happened - Arch has upgraded `python-cairocffi` to 1.6.0 and `python-xcffib` to 1.4.0. However, those are only supported in qtile-git and qtile v0.23.0 (which is not in the Arch repo yet).

So, either downgrade those two python packages or use the `qtile-git` package from the AUR and you should be fine.

**Comment by leemon22** [comment]
Solved downgrading python-cairocffi to 1.5.1-1 or 1.5.0-1 and python-xcffib to 1.2.0-2.
May it works installing qtile-git from the AUR, but I didn't try.
Thanks for the help
